pip/pip-321.md

```diff
@@ -0,0 +1,251 @@
+
+- Replication clusters configuration: 
+  - Set namespace `ns` replication clusters : cluster1, cluster2, cluster3 (local cluster); 
+  - Set topic `ns/topic1` replication clusters :  cluster1, cluster2, Cluster3; 
+  - Set topic `ns/topic2` replication clusters :  null;
+
+- Expected: 
+  - Topic `ns/change_event` can replication in cluster1, cluster2 and cluster3.
+
+- Actual: 
+  - Topic `ns/change_event` does not have a replication configuration, so the topic policies will not be replicated.
+
+### Solution 2: Introduce 'Topic-Policy-Synchronized-Clusters'
+
+For Problem2, we introduce 'topic-policy-synchronized-clusters' to configure the clusters where the topics under the namespace can replicate topic policies to.
+
+**Example:**
+
+- Environment: 
+  - cluster1, cluster2, and cluster3 in different regions using different Zookeeper clusters.
+
+- Replication clusters configuration: 
+  - Set namespace 'ns' replication clusters :  cluster1, cluster2, cluster3 (local cluster);
+  - Set topic `ns/topic1` replication clusters : cluster1, cluster2, cluster3;
+  - Set topic `ns/topic2` replication clusters :  null;
+  - Set `topic-policy-synchronized-clusters`:  cluster1, cluster2, cluster3;
+
+- Expected: 
+  - Topic `ns/change_event` can replicate in cluster1, cluster2 and cluster3.
+
+- Actual: 
+  - Topic policies can work as expected.
+
+**Note:** This is an optional operation. You can choose not to configure the 'Topic-Policy-Synchronized-Clusters' and define different topic policies in the different clusters.
+
+# Goals
+
+## In Scope
+
+The namespace has a more detailed configuration for clusters. Users can use `replication-clusters`, `allowed-clusters`, and `topic-policy-synchronized-clusters` to configure the different behavioral logic of topics in the multiple clusters, rather than just using `replication-clusters` to control all behaviors.
+
+## Out of Scope
+
+This proposal can be used to solve the problem of topic-level geo-replication can not work as expected. It is the initial motivation for this proposal, but this proposal does not involve modifications to geo-replication.
+
+# High Level Design
+
+Add two new namespace policies `allowed_clusters`, and `topic_policy_synchronized_clusters`.
+Replace the function of  `replication_clusters` with `replication_clusters`, `allowed_clusters`, and `topic_policy_synchronized_clusters`.
+
+# Detailed Design
+
+## Public-facing Changes
+
+### Public API
+Add a new endpoint `setNamespaceAllowedClusters` to the REST API.
+```
+Type: POST
+Path: /namespaces/{tenant}/{namespace}/allowedcluster
+Query parameters: tenant, namespace
+HTTP body parameters: List<String> clusterIds
+Response code:
+  * 400 Bad request, Allowed-clusters should contain all replication-clusters
+  * 403 Forbidden, Don't have admin permission
+  * 404 Not Found, Tenant or cluster or namespace doesn't exist.
+  * 409 Conflict, Peer-cluster can't be part of allowed-cluster.
```

Comment:
What does this mean?

Reply:
And it's also duplicated with the error code 412?

Reply:
There is an explanation below the line `code 412`. This follows the behavior of namespace replication clusters.
In my understanding, only one cluster in a `peer cluster group` can be a part of the `replication-clusters` at the same time.